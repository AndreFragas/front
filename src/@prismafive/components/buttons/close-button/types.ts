export interface CloseButtonProps {
  onClick?: () => void;
  color?: string;
}
