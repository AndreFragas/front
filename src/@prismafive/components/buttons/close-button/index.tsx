import {IconButton} from '@mui/material';
import React from 'react';
import Icon from 'src/@core/components/icon';
import {CloseButtonProps} from './types';

export const CloseButton = React.forwardRef<HTMLButtonElement, CloseButtonProps>((props, ref) => {
  const {onClick, color, ...otherProps} = props;

  return (
    <IconButton {...otherProps} ref={ref} onClick={props.onClick} data-testid="CloseButton">
      <Icon icon="tabler:x" fontSize={20} color={props.color} />
    </IconButton>
  );
});
