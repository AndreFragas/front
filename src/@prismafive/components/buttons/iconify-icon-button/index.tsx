import {IconButton} from '@mui/material';
import Icon from 'src/@core/components/icon';
import {IconifyIconButtonProps} from './types';

export function IconifyIconButton(props: IconifyIconButtonProps) {
  return (
    <IconButton onClick={props.onClick} data-testid="CloseButton" className={props.className}>
      <Icon icon={props.icon} fontSize={20} color={props.color} />
    </IconButton>
  );
}
