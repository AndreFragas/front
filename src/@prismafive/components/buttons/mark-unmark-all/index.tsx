import {Button, Tooltip} from '@mui/material';
import {lowercaseExceptFirstLetter} from 'src/@prismafive/helper/formatters';
import {useTranslate} from 'src/@prismafive/hooks/use-translate';
import {MarkUnmarkAllButtonProps} from './types';

export function MarkUnmarkAllButton(props: MarkUnmarkAllButtonProps) {
  const {translate} = useTranslate();

  function renderButton() {
    return (
      <Button variant="outlined" onClick={props.markAllState ? props.unmarkAll : props.markAll} sx={props.sx}>
        {props.markAllState
          ? lowercaseExceptFirstLetter(
              props.customUnmarkText ? translate(props.customUnmarkText) : translate('global.unmarkAll')
            )
          : lowercaseExceptFirstLetter(
              props.customMarkText ? translate(props.customMarkText) : translate('global.markAll')
            )}
      </Button>
    );
  }
  const markTooltipText = props.customMarkTooltipText
    ? translate(props.customMarkTooltipText)
    : translate('global.markAll');

  const unmarkTooltipText = props.customUnmarkTooltipText
    ? translate(props.customUnmarkTooltipText)
    : translate('global.unmarkAll');

  return (
    <>
      {props.disableTooltip ? (
        renderButton()
      ) : (
        <Tooltip title={props.markAllState ? unmarkTooltipText : markTooltipText}>{renderButton()}</Tooltip>
      )}
    </>
  );
}
