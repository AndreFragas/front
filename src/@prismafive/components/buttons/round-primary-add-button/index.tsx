import {useTheme} from '@mui/material/styles';
import Icon from 'src/@core/components/icon';
import {blackOrWhite} from 'src/@prismafive/helper/black-or-white';
import {RoundPrimaryAddButtonProps} from './types';

export function RoundPrimaryAddButton(props: RoundPrimaryAddButtonProps) {
  const theme = useTheme();
  return (
    <button
      style={{
        width: 40,
        height: 40,
        borderRadius: '100%',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        border: 'none',
        backgroundColor: theme.palette.primary.main,
        cursor: 'pointer',
        ...(props.margin && {margin: props.margin}),
      }}
      onClick={() => props.onClick && props.onClick()}
    >
      <Icon icon={'ic:baseline-plus'} fontSize={25} color={blackOrWhite(theme.palette.primary.main)} />
    </button>
  );
}
