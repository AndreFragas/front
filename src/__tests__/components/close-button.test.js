import {cleanup, fireEvent, render, screen, waitFor} from '@testing-library/react';
import renderer from 'react-test-renderer';
import {CloseButton} from 'src/@prismafive/components/buttons/close-button';

afterEach(() => {
  cleanup();
});

test('execute onClick when clicked', () => {
  render(<CloseButton onClick={() => console.log('clicked!')} />);
  const consoleLogSpy = jest.spyOn(console, 'log').mockImplementation();
  const element = screen.getByTestId('CloseButton');
  expect(element).toBeInTheDocument();

  fireEvent.click(element);

  expect(consoleLogSpy).toHaveBeenCalledWith('clicked!');

  expect(element).toContainHTML('<button');
});

test('matches snapshot', () => {
  const tree = renderer.create(<CloseButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
