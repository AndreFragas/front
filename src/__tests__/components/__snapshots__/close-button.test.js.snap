// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
  data-testid="CloseButton"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span />
</button>
`;
